Scala.js backend fails with "Invalid name" for closure parameter name containing "["

After migrating a code base to scala.js 1.5.1  and scala3 `compileIncremental` phase fails with error:
```
error] java.lang.IllegalArgumentException: Invalid name: [-3, 3]$2
[error] 	at org.scalajs.ir.Names$.org$scalajs$ir$Names$$$throwInvalidEncodedName(Names.scala:536)
[error] 	at org.scalajs.ir.Names$.org$scalajs$ir$Names$$$validateSimpleEncodedName(Names.scala:550)
[error] 	at org.scalajs.ir.Names$.org$scalajs$ir$Names$$$validateSimpleEncodedName(Names.scala:539)
[error] 	at org.scalajs.ir.Names$LocalName$.apply(Names.scala:106)
[error] 	at org.scalajs.ir.Names$LocalName$.apply(Names.scala:109)
[error] 	at dotty.tools.backend.sjs.JSEncoding$LocalNameGenerator.freshLocalIdent(JSEncoding.scala:109)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.freshLocalIdent(JSCodeGen.scala:132)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.$anonfun$29(JSCodeGen.scala:3044)
[error] 	at scala.collection.immutable.List.map(List.scala:250)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genClosure(JSCodeGen.scala:3051)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr(JSCodeGen.scala:1532)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr(JSCodeGen.scala:1469)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genExpr(JSCodeGen.scala:1263)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genActualArgs$$anonfun$1(JSCodeGen.scala:3679)
[error] 	at scala.collection.immutable.List.map(List.scala:250)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genActualArgs(JSCodeGen.scala:3679)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genNormalApply(JSCodeGen.scala:2635)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genApply(JSCodeGen.scala:1722)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr(JSCodeGen.scala:1390)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr(JSCodeGen.scala:1469)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr(JSCodeGen.scala:1469)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr(JSCodeGen.scala:1469)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr(JSCodeGen.scala:1469)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStat(JSCodeGen.scala:1241)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr$$anonfun$5(JSCodeGen.scala:1469)
[error] 	at scala.collection.immutable.List.map(List.scala:246)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStatOrExpr(JSCodeGen.scala:1469)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genStat(JSCodeGen.scala:1241)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genBody$1(JSCodeGen.scala:1145)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genMethodDef(JSCodeGen.scala:1151)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genMethodWithCurrentLocalNameScope$$anonfun$1(JSCodeGen.scala:1118)
[error] 	at dotty.tools.backend.sjs.ScopedVar$.withScopedVars(ScopedVar.scala:35)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genMethodWithCurrentLocalNameScope(JSCodeGen.scala:1124)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genMethod$$anonfun$1(JSCodeGen.scala:1022)
[error] 	at dotty.tools.backend.sjs.ScopedVar$.withScopedVars(ScopedVar.scala:35)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genMethod(JSCodeGen.scala:1023)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genScalaClass$$anonfun$2(JSCodeGen.scala:341)
[error] 	at scala.collection.immutable.List.foreach(List.scala:333)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genScalaClass(JSCodeGen.scala:346)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genCompilationUnit$$anonfun$6$$anonfun$1(JSCodeGen.scala:217)
[error] 	at dotty.tools.backend.sjs.ScopedVar$.withScopedVars(ScopedVar.scala:35)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genCompilationUnit$$anonfun$2(JSCodeGen.scala:221)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.immutable.List.foreach(List.scala:333)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.genCompilationUnit(JSCodeGen.scala:223)
[error] 	at dotty.tools.backend.sjs.JSCodeGen.run(JSCodeGen.scala:152)
[error] 	at dotty.tools.backend.sjs.GenSJSIR.run(GenSJSIR.scala:15)
[error] 	at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:303)
[error] 	at scala.collection.immutable.List.map(List.scala:250)
[error] 	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:304)
[error] 	at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:205)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[error] 	at dotty.tools.dotc.Run.runPhases$5(Run.scala:215)
[error] 	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:223)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[error] 	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
[error] 	at dotty.tools.dotc.Run.compileUnits(Run.scala:230)
[error] 	at dotty.tools.dotc.Run.compileSources(Run.scala:166)
[error] 	at dotty.tools.dotc.Run.compile(Run.scala:150)
[error] 	at dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
[error] 	at dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] 	at dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] 	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:186)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:241)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:176)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:157)
[error] 	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:157)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:204)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:174)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:172)
[error] 	at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:457)
[error] 	at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] 	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:261)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:412)
[error] 	at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:499)
[error] 	at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:399)
[error] 	at sbt.internal.inc.Incremental$.apply(Incremental.scala:166)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:482)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] 	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2346)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2303)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] 	at sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2299)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
```

Unfortently I have no idea how to make minimised version.

@smarter unfortunately I can't shared the code :(

But I definitely can recompile it with `-Ycheck:all`. Here the new log:
```
[error] java.lang.AssertionError: assertion failed: non member selection of method asInstanceOf in class Object from (ev$1 : ErasedValueType(scala.scalajs.js.UndefOrOps, Object)) in ev$1.asInstanceOf
[error] 	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[error] 	at dotty.tools.dotc.transform.FirstTransform.checkPostCondition(FirstTransform.scala:59)
[error] 	at dotty.tools.dotc.transform.TreeChecker.dotty$tools$dotc$transform$TreeChecker$Checker$$_$typedUnadapted$$anonfun$1(TreeChecker.scala:341)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.immutable.List.foreach(List.scala:333)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:341)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2916)
[error] 	at dotty.tools.dotc.typer.Applications.typedTypeApply(Applications.scala:1073)
[error] 	at dotty.tools.dotc.typer.Applications.typedTypeApply$(Applications.scala:303)
[error] 	at dotty.tools.dotc.typer.Typer.typedTypeApply(Typer.scala:103)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2688)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$2(ProtoTypes.scala:414)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:370)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:415)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:836)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:836)
[error] 	at dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:530)
[error] 	at dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:595)
[error] 	at dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:433)
[error] 	at dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:721)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:835)
[error] 	at dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1046)
[error] 	at dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:303)
[error] 	at dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:103)
[error] 	at dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:893)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$5$$anonfun$4(Applications.scala:972)
[error] 	at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:2924)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:983)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1021)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:303)
[error] 	at dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:103)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2669)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2916)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1018)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock$$anonfun$2$$anonfun$1(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withDefinedSyms(TreeChecker.scala:190)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock$$anonfun$1(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withBlock(TreeChecker.scala:218)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2677)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedTyped(ReTyper.scala:62)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2674)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2916)
[error] 	at dotty.tools.dotc.typer.Typer.typedValDef(Typer.scala:2078)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2643)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2730)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2822)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2872)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedStats(TreeChecker.scala:524)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1012)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1016)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock$$anonfun$2$$anonfun$1(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withDefinedSyms(TreeChecker.scala:190)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock$$anonfun$1(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withBlock(TreeChecker.scala:218)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2677)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.typedLabeled(Typer.scala:1613)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedLabeled$$anonfun$1(TreeChecker.scala:529)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withDefinedSyms(TreeChecker.scala:190)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedLabeled(TreeChecker.scala:529)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2662)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2730)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2916)
[error] 	at dotty.tools.dotc.typer.Typer.$anonfun$34(Typer.scala:2142)
[error] 	at dotty.tools.dotc.typer.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:225)
[error] 	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2142)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedDefDef$$anonfun$1(TreeChecker.scala:481)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withDefinedSyms(TreeChecker.scala:190)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedDefDef(TreeChecker.scala:484)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2646)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2730)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2822)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2872)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedStats(TreeChecker.scala:524)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1012)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1016)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock$$anonfun$2$$anonfun$1(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withDefinedSyms(TreeChecker.scala:190)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock$$anonfun$1(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withBlock(TreeChecker.scala:218)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2677)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$2(ProtoTypes.scala:414)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:370)
[error] 	at dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:415)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:836)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:836)
[error] 	at dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:530)
[error] 	at dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:595)
[error] 	at dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:595)
[error] 	at dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:433)
[error] 	at dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:721)
[error] 	at dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:835)
[error] 	at dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1046)
[error] 	at dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:303)
[error] 	at dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:103)
[error] 	at dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:893)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$5$$anonfun$4(Applications.scala:972)
[error] 	at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:2924)
[error] 	at dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:983)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1021)
[error] 	at dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:303)
[error] 	at dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:103)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2669)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2849)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2872)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedStats(TreeChecker.scala:524)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1012)
[error] 	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1016)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock$$anonfun$2$$anonfun$1(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withDefinedSyms(TreeChecker.scala:190)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock$$anonfun$1(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withBlock(TreeChecker.scala:218)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedBlock(TreeChecker.scala:506)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2677)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2916)
[error] 	at dotty.tools.dotc.typer.Typer.$anonfun$34(Typer.scala:2142)
[error] 	at dotty.tools.dotc.typer.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:225)
[error] 	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2142)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedDefDef$$anonfun$1(TreeChecker.scala:481)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.withDefinedSyms(TreeChecker.scala:190)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedDefDef(TreeChecker.scala:484)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2646)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2730)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2822)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2872)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedStats(TreeChecker.scala:524)
[error] 	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2328)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedClassDef(TreeChecker.scala:459)
[error] 	at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$2(Typer.scala:2657)
[error] 	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2661)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2730)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2822)
[error] 	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2872)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedStats(TreeChecker.scala:524)
[error] 	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2451)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedPackageDef(TreeChecker.scala:550)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2702)
[error] 	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2731)
[error] 	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:118)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typedUnadapted(TreeChecker.scala:324)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2796)
[error] 	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2800)
[error] 	at dotty.tools.dotc.transform.TreeChecker$Checker.typed(TreeChecker.scala:308)
[error] 	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2916)
[error] 	at dotty.tools.dotc.transform.TreeChecker.check(TreeChecker.scala:142)
[error] 	at dotty.tools.dotc.transform.TreeChecker.run(TreeChecker.scala:111)
[error] 	at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:303)
[error] 	at scala.collection.immutable.List.map(List.scala:250)
[error] 	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:304)
[error] 	at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:205)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] 	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] 	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[error] 	at dotty.tools.dotc.Run.runPhases$5(Run.scala:215)
[error] 	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:223)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[error] 	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
[error] 	at dotty.tools.dotc.Run.compileUnits(Run.scala:230)
[error] 	at dotty.tools.dotc.Run.compileSources(Run.scala:166)
[error] 	at dotty.tools.dotc.Run.compile(Run.scala:150)
[error] 	at dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
[error] 	at dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] 	at dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] 	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:186)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:241)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:176)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:157)
[error] 	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:157)
[error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:204)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:174)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:172)
[error] 	at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:457)
[error] 	at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] 	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:261)
[error] 	at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:412)
[error] 	at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:499)
[error] 	at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:399)
[error] 	at sbt.internal.inc.Incremental$.apply(Incremental.scala:166)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:528)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:482)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[error] 	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] 	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2346)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2303)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] 	at sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2299)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
```

--- a/compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala
+++ b/compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala
@@ -3041,7 +3041,7 @@ class JSCodeGen()(using genCtx: Context) {
     val formalAndActualCaptures = allCaptureValues.map { value =>
       implicit val pos = value.span
       val (formalIdent, originalName) = value match {
-        case Ident(name) => (freshLocalIdent(name.toString), OriginalName(name.toString))
+        case Ident(name) => (freshLocalIdent(name.toTermName), OriginalName(name.toString))
         case This(_)     => (freshLocalIdent("this"), thisOriginalName)
         case _           => (freshLocalIdent(), NoOriginalName)
       }
@@ -3069,7 +3069,7 @@ class JSCodeGen()(using genCtx: Context) {
 
     val formalAndActualParams = formalParamNames.lazyZip(formalParamTypes).lazyZip(formalParamRepeateds).map {
       (name, tpe, repeated) =>
-        val formalParam = js.ParamDef(freshLocalIdent(name.toString),
+        val formalParam = js.ParamDef(freshLocalIdent(name),
             OriginalName(name.toString), jstpe.AnyType, mutable = false)
         val actualParam =
           if (repeated) genJSArrayToVarArgs(formalParam.ref)(tree.sourcePos)

--- a/tests/sjs-junit/test/org/scalajs/testsuite/compiler/RegressionTestScala3.scala
+++ b/tests/sjs-junit/test/org/scalajs/testsuite/compiler/RegressionTestScala3.scala
@@ -27,6 +27,14 @@ class RegressionTestScala3 {
     assertEquals("foo", new RangeErrorIssue11592("foo").message)
     assertEquals("", new RangeErrorIssue11592().message)
   }
+
+  @Test def testNonJVMCharsInClosureParametersIssue12507(): Unit = {
+    def foo(`[-3, 3]`: Int): Int => Int = { x =>
+      `[-3, 3]`
+    }
+
+    assertEquals(5, foo(5)(4))
+  }
 }
 
 object RegressionTestScala3 {

